src/components/semantic/props/CheckboxDocProp.tsx

```diff
@@ -9,18 +9,31 @@ type CheckboxDocProps<K extends keyof D, D> =
     prop: K;
     label: string;
     checkedIfUndefined?: boolean;
+    disabled?: boolean;
 };
 
 export function CheckboxDocProp<K extends keyof D, D extends { [Key in K]?: boolean }>({
     doc,
     update,
     prop,
     label,
-    checkedIfUndefined
+    checkedIfUndefined,
+    disabled,
 }: CheckboxDocProps<K, D>) {
-    return <Label className={styles.checkboxLabel}>
+    useEffect(()=> {
+        if (disabled && doc[prop]) {
+            update({
+                ...doc,
+                [prop]: false,
+            });
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [disabled, doc[prop]]);
```

Comment:
On a related note, I'm not sure I understand the purpose of `doc[prop]` in the dependency here -- if we only do anything when `disabled` is true, we wouldn't be able to change the prop anyway?